stm32-modules/heater-shaker/firmware/system/system_hardware.h

```diff
@@ -1,16 +1,65 @@
 #ifndef SYSTEM_HARDWARE_H__
 #define SYSTEM_HARDWARE_H__
 #include "stm32f3xx_hal_gpio.h"
+#include "stm32f3xx_hal.h"
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
+#include <stdbool.h>
+#include "systemwide.h"
+
+#define I2C_ADDRESS        0x6C
+/* I2C TIMING Register is defined when I2C clock source is SYSCLK (true) */
+/* I2C TIMING is calculated using SYSCLK = 72 MHz, I2C Speed Frequency = 100 KHz, Rise Time = 100ns, and Fall Time = 100ns */
+#define I2C_TIMING      0x00201D2B
+
+/* Definition for I2Cx clock resources */
+#define I2Cx                            I2C1
+#define RCC_PERIPHCLK_I2Cx              RCC_PERIPHCLK_I2C1
+#define RCC_I2CxCLKSOURCE_SYSCLK        RCC_I2C1CLKSOURCE_SYSCLK
+#define I2Cx_CLK_ENABLE()               __HAL_RCC_I2C1_CLK_ENABLE()
```

Comment:
having these in this header, for instance, will probably cause compiler errors when it's included from c++